engine/access/rest/handlers.go

```diff
@@ -0,0 +1,228 @@
+	// create h logger for the request
+	errorLogger := h.logger.With().Str("request_url", r.URL.String()).Logger()
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	vars := mux.Vars(r)
+	idParam := vars["id"]
+
+	// gorilla mux retains opening and ending square brackets for ids
+	idParam = strings.TrimSuffix(idParam, "]")
+	idParam = strings.TrimPrefix(idParam, "[")
+
+	ids := strings.Fields(idParam)
+
+	blocks := make([]*generated.Block, len(ids))
+
+	if len(ids) > MaxAllowedBlockIDsCnt {
+		h.errorResponse(w, http.StatusBadRequest, fmt.Sprintf("at most %d Block IDs can be requested at a time", MaxAllowedBlockIDsCnt), errorLogger)
+		return
+	}
+
+	for i, id := range ids {
+		flowID, err := flow.HexStringToIdentifier(id)
+		if err != nil {
+			h.errorResponse(w, http.StatusBadRequest, fmt.Sprintf("invalid ID %s", id), errorLogger)
+			return
+		}
+
+		flowBlock, err := h.backend.GetBlockByID(r.Context(), flowID)
+		if err != nil {
+			// if error has GRPC code NotFound, the return HTTP NotFound error
+			if status.Code(err) == codes.NotFound {
+				h.errorResponse(w, http.StatusNotFound, fmt.Sprintf("block with ID %s not found", id), errorLogger)
+				return
+			}
+			errorLogger.Error().Err(err).Str("block_id", id).Msg("failed to look up block")
+			h.errorResponse(w, http.StatusInternalServerError, fmt.Sprintf("failed to look up block with ID %s", id), errorLogger)
+			return
+		}
+		blocks[i] = blockResponse(flowBlock)
+	}
+
+	h.jsonResponse(w, blocks, errorLogger)
+}
+
+// GetTransactionByID gets a transaction by requested ID.
+func (h *Handlers) GetTransactionByID(w http.ResponseWriter, r *http.Request) {
+	errorLogger := h.logger.With().Str("request_url", r.URL.String()).Logger() // todo(sideninja) refactor this to be initialized for us
+
+	vars := mux.Vars(r)
+	idFromRequest := vars["id"]
+	id, err := toID(idFromRequest)
+	if err != nil {
+		h.errorResponse(w, http.StatusBadRequest, fmt.Sprintf("invalid ID: %s", idFromRequest), errorLogger)
+		return
+	}
+
+	tx, err := h.backend.GetTransaction(r.Context(), id)
+	if err != nil {
+		h.errorResponse(w, http.StatusBadRequest, fmt.Sprintf("transaction fetching error: %s", err.Error()), errorLogger)
+		return
+	}
+
+	h.jsonResponse(w, transactionResponse(tx), errorLogger)
+}
+
+// CreateTransaction creates a new transaction from provided payload.
+func (h *Handlers) CreateTransaction(w http.ResponseWriter, r *http.Request) {
+	var txBody generated.TransactionsBody
+	err := h.jsonDecode(r.Body, &txBody)
+	if err != nil {
+		h.errorResponse(w, http.StatusBadRequest, err.Error(), h.logger)
+		return
+	}
+
+	tx, err := toTransaction(&txBody)
+	if err != nil {
+		h.errorResponse(w, http.StatusBadRequest, err.Error(), h.logger) // todo(sideninja) use refactor err func
+		return
+	}
+
+	err = h.backend.SendTransaction(r.Context(), &tx)
+	if err != nil {
+		h.errorResponse(w, http.StatusBadRequest, err.Error(), h.logger)
```

Comment:
@sideninja actually has a follow up PR which addresses error handling (translating GRPC errors to HTTP errors more accurately). https://github.com/onflow/flow-go/pull/1640/files#diff-32628f0bc7adc671dce50f45fe5e5ad51e5c0b2c0f3c06dcbae6d866bbaf5a52R76.
That should address the error reporting (or if not then I will fix it over there since a lot of this will be changing on the next PR)